libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/ptx/ptx_helper_functions.h

```diff
@@ -46,15 +46,19 @@ inline _LIBCUDACXX_DEVICE _CUDA_VSTD::uint64_t __as_ptr_gmem(const void* __ptr)
 template <typename _Tp>
 inline _LIBCUDACXX_DEVICE _CUDA_VSTD::uint32_t __as_b32(_Tp __val)
 {
+#if _CCCL_STD_VER >= 2017
```

Comment:
The reason we cannot have below static_assert is that this function is sometimes called in an 
`if _LIBCUDACXX_CONSTEXPR_AFTER_CXX14(condition) {__as_b32(...)}` block.

The `_LIBCUDACXX_CONSTEXPR_AFTER_CXX14` macro is defined as:
```cuda
#if _CCCL_STD_VER > 2014 && !defined(_LIBCUDACXX_HAS_NO_CXX14_CONSTEXPR)
#  define _LIBCUDACXX_CONSTEXPR_AFTER_CXX14 constexpr
#else
#  define _LIBCUDACXX_CONSTEXPR_AFTER_CXX14
#endif
```

I don't know to which extent there are C++17 compilers that do not implement `if constexpr`, but if there are, then the following check would avoid more cases slipping through:
```cuda
#if _CCCL_STD_VER > 2014 && !defined(_LIBCUDACXX_HAS_NO_CXX14_CONSTEXPR)
```